### PR TITLE
Code style change

### PR DESCRIPTION
**Pull request by cpsw**
The code changes make the control flow logic explicit, as the second expression of && is evaluated only when the first one is evaluated to true.

### CODE REVIEW

File: stream/stream.c [hunk 1]
```diff
@@ -569,7 +569,10 @@ bool stream_seek(stream_t *s, int64_t pos)
     }
 
     if (s->mode == STREAM_WRITE)
-        return s->seekable && s->seek(s, pos);
+        if (s->seekable)
+            return s->seek(s, pos);
+        else
+            return s->seekable;
```
Comment (sfan5) [inline stream/stream.c]: == `return false;`

**Comment by rossy** [comment]
Disagree. The short-circuiting behaviour of logical operators is already pretty common knowledge IMO.

**Comment by wm4** [comment]
As a cosmetic change it probably wouldn't be that bad to make the code a bit less tricky and more explicit, but the new code still doesn't follow the project coding style. Is it really that hard to read contribute.md?

**Comment by cpsw** [comment]
Sorry about that, @wm4. I'm new to this world. I just added the brackets to follow the project's guidelines.

File: stream/stream.c (hunk 1 above)
Comment (cpsw) [reply stream/stream.c]: Hi @sfan5, do you mean (s->seekable == true) and return false in the else branch?

**Comment by SirCmpwn** [comment]
I agree with rossy, -1

File: stream/stream.c (hunk 1 above)
Comment (sfan5) [reply stream/stream.c]: I mean that the else branch is equivalent to `return false;` so you should just make the explicit.

**Comment by jeeb** [comment]
First of all, thanks for trying to contribute.

But yes, I am not sure if I agree with the necessity of this. It's a C `bool` returned from a classic `X && Y`, thing where the first one is meant to be a controller of the second one. Business as usual as far as I am concerned.

Also, even the version posted by @cpsw IMHO isn't too great since while you've now got `{}` around the if/else, suddenly the first `if` level is now multi-line but without braces. In my opinion that would be a much more useful place to have braces at than the single-line internal if/else.
```c
    if (s->mode == STREAM_WRITE) {
        if (s->seekable)
            return s->seek(s, pos);
        else
            return s->seekable;
    }
```

In other words, sorry but no bazooka this time.

**Comment by haasn** [comment]
Just to throw some more paint on the bike-shed, here would be one way of writing it in a more “clear” way:

```
return s->seekable ? s->seek(s, pos) : false;
```

but I agree, `cond && action()` is a common idiom in C. You'll get used to it.
